grr/core/grr_response_core/lib/config_lib.py

```diff
@@ -1532,6 +1532,18 @@ def DEFINE_integer_list(self, name, default, help, constant=False):
             validator=type_info.Integer()),
         constant=constant)
 
+  def DEFINE_semantic_value_list(self, semantic_type, name, default=None, help=""):
+    if issubclass(semantic_type, rdf_structs.RDFStruct):
+      raise ValueError("DEFINE_semantic_value_list should be used for types "
+                       "based on RDFValues.")
+    self.AddOption(
+        type_info.RDFValueList(
+            rdfclass=semantic_type,
+            name=name,
+            default=default,
+            validator=type_info.RDFValueType(rdfclass=semantic_type),
+            description=help))
+
```

Comment:
That, on the other hand, I'm really not sure how to tackle.
The only way I know of to represent a cert chain is to actually simply concatenate all the involved certs (think /etc/ssl/certs/ca-certificates.crt).

Which basically makes it a list of x509 certificates, hence that semantic_value_list (which, admittedly, is an horrendous hack which I'm not proud of).

So, could you provide me with some pointers on what exactly is this canonical text representation you're mentioning? Then I'll be glad to implement the requested change and get rid of the list :)

Reply:
Sorry for a delay with the reply. By canonical text representation I indeed mean a simple concatenation of all the certs.

In your current PR you expect a list of certificates in the configuration and then you concatenate them and pass to the communication layer. But GRR doesn't need to access each individual certificate, it will concatenate them anyway and delegate everything to the "requests" library. With a type like a suggested RDFX509CertChain, you can provide the certificates chain as a concatenated string right in the config and then simply store it in a file and and pass further. 

To be fair, I'd also see no problem with simply using a string type and DEFINE_string for Client.transport_ca_certificate. You can then simply read the chain from the config option, put it into a temp file and pass it on.

What do you think?